keps/sig-apps/3329-retriable-and-non-retriable-failures/README.md

```diff
@@ -0,0 +1,1543 @@
+- Pod status:
+  - status: Terminating
+  - `phase=Failed`
+  - `reason=`
+  - `message=`
+- Container status:
+  - `state=Ternminated`
+  - `exitCode=137`
+  - `reason=Error`
+- Retriable: Yes
+
+##### Taint-based eviction
+
+- Reproduction: We run a long-running job. Then, we taint the node with `NoExecute`
+- Comments: controlled by kube-scheduler in `controller/nodelifecycle/scheduler/taint_manager.go`
+- Pod status:
+  - status: Terminating
+  - `phase=Failed`
+  - `reason=`
+  - `message=`
+- Container status:
+  - `state=Ternminated`
+  - `exitCode=137`
+  - `reason=Error`
+- Retriable: Yes
+
+##### Node-pressure eviction
+
+Memory-pressure eviction:
+
+- Reproduction: We run a job with a pod which attempts to allocate more
+  memory than available on the node
+- Comments: controlled by kubelet in `kubelet/eviction/eviction_manager.go`
+- Pod status:
+  - status: ContainerStatusUnknown
+  - `phase=Failed`
+  - `reason=Evicted`
+  - `message=The node was low on resource: memory. (...)`
+- Container status:
+  - `state=Ternminated`
+  - `exitCode=137`
+  - `reason=ContainerStatusUnknown`
+- Retriable: Unclear, excessive memory usage suggests a bug or misconfiguration.
+  However, a restart on another node may succeed
+
+Disk-pressure eviction:
+
+- Reproduction: We run a job with a pod which attempts to write more
+  data than the disk space available on the node
+- Comments: controlled by kubelet in `kubelet/eviction/eviction_manager.go`
+- Pod status:
+  - status: Error
+  - `phase=Failed`
+  - `reason=Evicted`
+  - `message=The node was low on resource: ephemeral-storage. (...)`
+- Container status:
+  - `state=Ternminated`
+  - `exitCode=137`
+  - `reason=Error`
+- Retriable: Unclear, excessive disk usage suggests a bug or misconfiguration.
+  However, a restart on another node may succeed
+
+##### OOM kill
+
+- Reproduction: We run a job with a pod which attempts to allocate more
+  memory than constrained in the container spec by `resources.limits.memory`
+- Comments: handled by kubelet
+- Pod status:
+  - status: OOMKilled
+  - `phase=Failed`
+  - `reason=`
+  - `message=`
+- Container status:
+  - `state=Ternminated`
+  - `exitCode=137`
+  - `reason=OOMKilled`
+- Retriable: Unclear, but if occurs when
+  `resources.requests.memory=resources.limits.memory` it strongly suggests
+  a software bug.
+
+##### Node drain
+
+- Reproduction: We run a job with a long-running pod, then drain the node
+  with the `kubectl drain` command
+- Comments: performed by Eviction API
+- Pod status:
+  - status: Terminating
+  - `phase=Failed`
+  - `reason=`
+  - `message=`
+- Container status:
+  - `state=Ternminated`
+  - `exitCode=137`
+  - `reason=Error`
+- Retriable: Yes
+
+##### Disconnected node
+
+- Reproduction: We run a job with a long-running pod, then disconnect the node
+  and delete it by the `kubectl delete` command
+- Comments: unclear handling, the pod remains in the `Running` phase. Potentially,
```

Comment:
Or investigate why it is not moved to Failed phase and "fix" - but it seems outside the KEP. 

Alternatively, just leave this scenario as outside the scope of the KEP and focus on those which are in Failed phase
only, because I think matching running phase could be complicated in general.